engine_modules/invisible_hand/tasks.py

```diff
@@ -21,13 +23,15 @@ def run(self, game):
 		if len(corpos) == 0:
 			return
 
-		corpos[0].update_assets(1, category=AssetDelta.INVISIBLE_HAND)
-		content = u'La main du marché favorise le marché historique de la corpo %s.' % (corpos[0].base_corporation.name)
-		game.add_newsfeed(category=Newsfeed.ECONOMY, content=content, status=Newsfeed.PRIVATE, market=corpos[0].historic_market, corporations=[corpos[0]])
+		market = random.choice(corpos[0].markets)
```

Comment:
i think you should add a random_market property on corporation, to use here and with the effect function